### ISSUE TITLE
cannot deploy openshift 3.7 - glusterfs issue

### ISSUE THREAD

**Issue by pgeorgiev333** [openshift/openshift-ansible#6339]
#### Description

Trying to install openshift 3.7 from the master branch but it fails on the glusterfs part

##### Version

```
ansible 2.4.0.0
  config file = /root/openshift-ansible/ansible.cfg
  configured module search path = [u'/root/openshift-ansible/library']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.5 (default, Aug  4 2017, 00:39:18) [GCC 4.8.5 20150623 (Red Hat 4.8.5-16)]

git describe
openshift-ansible-3.8.0-0.13.0-75-gb62ec2d

```

##### Steps To Reproduce
1. Configure and run the playbook:

```
ansible-playbook playbooks/byo/config.yml
```

##### Expected Results
Installation IS OK

##### Observed Results
Installation fails

```
fatal: [openshift.xxx]: FAILED! => {"changed": false, "failed": true, "module_stderr": "Shared connection to openshift.xxx closed.\r\n", "module_stdout": "Traceback (most recent call last):\r\n  File \"/tmp/ansible_u8foww/ansible_module_oc_process.py\", line 1635, in <module>\r\n    main()\r\n  File \"/tmp/ansible_u8foww/ansible_module_oc_process.py\", line 1628, in main\r\n    rval = OCProcess.run_ansible(module.params, module.check_mode)\r\n  File \"/tmp/ansible_u8foww/ansible_module_oc_process.py\", line 1584, in run_ansible\r\n    all_results = ocprocess.needs_update()\r\n  File \"/tmp/ansible_u8foww/ansible_module_oc_process.py\", line 1503, in needs_update\r\n    for obj in self.template:\r\n  File \"/tmp/ansible_u8foww/ansible_module_oc_process.py\", line 1449, in template\r\n    raise OpenShiftCLIError('Error processing template [%s]: %s' %(self.name, results))\r\n__main__.OpenShiftCLIError: Error processing template [glusterfs]: {'cmd': '/usr/bin/oc process glusterfs -v CLUSTER_NAME=storage NODE_LABELS={\"glusterfs\": \"storage-host\"} IMAGE_VERSION=latest IMAGE_NAME=gluster/gluster-centos GB_GLFS_LRU_COUNT=15 -n glusterfs', 'returncode': 1, 'results': {}, 'stderr': u'Flag --value has been deprecated, Use -p, --param instead.\\nerror: unknown parameter name \"GB_GLFS_LRU_COUNT\"\\n', 'stdout': u''}\r\n", "msg": "MODULE FAILURE", "rc": 0}
```

##### Additional Information

* CentOS 7

```
[OSEv3:children]
masters
nodes
etcd
nfs
glusterfs

[OSEv3:vars]

openshift_deployment_type=origin

ansible_ssh_user=root
containerized=false
openshift_master_cluster_hostname=openshift.xxx
openshift_master_cluster_public_hostname=openshift.xxx

# Can set multitenant plugin
os_sdn_network_plugin_name=redhat/openshift-ovs-subnet

openshift_master_identity_providers=[{'name': 'htpasswd_auth', 'login': 'true', 'challenge': 'true', 'kind': 'HTPasswdPasswordIdentityProvider', 'filename': '/etc/origin/master/htpasswd'}]

os_firewall_use_firewalld=true

openshift_portal_net=172.30.0.0/16
openshift_master_default_subdomain=xxx

openshift_node_proxy_mode=iptables

osm_cluster_network_cidr=10.128.0.0/14

osm_host_subnet_length=9

openshift_docker_additional_registries=docker.io,yy.yy.yy.yy:5000
openshift_docker_insecure_registries=yy.yy.yy.yy:5000

# Metrics
openshift_metrics_install_metrics=true

openshift_metrics_hawkular_hostname=metrics.xxx
openshift_logging_kibana_hostname=kibana.xxx

openshift_metrics_storage_kind=nfs
openshift_metrics_storage_access_modes=['ReadWriteOnce']
openshift_metrics_storage_nfs_directory=/exports
openshift_metrics_storage_nfs_options='*(rw,root_squash)'
openshift_metrics_storage_volume_name=metrics
openshift_metrics_storage_volume_size=10Gi

openshift_deployment_type=origin

openshift_hosted_registry_storage_kind=nfs
openshift_hosted_registry_storage_access_modes=['ReadWriteMany']
openshift_hosted_registry_storage_nfs_directory=/exports
openshift_hosted_registry_storage_nfs_options='*(rw,root_squash)'
openshift_hosted_registry_storage_volume_name=registry
openshift_hosted_registry_storage_volume_size=10Gi

openshift_storage_glusterfs_namespace=glusterfs
openshift_storage_glusterfs_name=storage

openshift_logging_install_logging=true

openshift_hosted_etcd_storage_kind=nfs
openshift_hosted_etcd_storage_nfs_options="*(rw,root_squash,sync,no_wdelay)"
openshift_hosted_etcd_storage_nfs_directory=/exports 
openshift_hosted_etcd_storage_volume_name=etcd-vol2 
openshift_hosted_etcd_storage_access_modes=["ReadWriteOnce"]
openshift_hosted_etcd_storage_volume_size=1G
openshift_hosted_etcd_storage_labels={'storage': 'etcd'}
ansible_service_broker_image_prefix=openshift/
ansible_service_broker_registry_url="registry.access.redhat.com"

[masters]
openshift.xxx

[nodes]
openshift.xxx openshift_schedulable=true openshift_node_labels="{'region': 'infra'}"
oc-node-1.xxx openshift_node_labels="{'region': 'primary', 'zone': 'bg'}"
oc-node-2.xxx openshift_node_labels="{'region': 'primary', 'zone': 'bg'}"
oc-gluster-1.xxx
oc-gluster-2.xxx
oc-gluster-3.xxx

[etcd]
openshift.xxx

[nfs]
openshift.xxx

[glusterfs]
oc-gluster-1.xxx glusterfs_ip=10.232.3.56 glusterfs_devices='[ "/dev/sdc" ]'
oc-gluster-2.xxx glusterfs_ip=10.232.3.57 glusterfs_devices='[ "/dev/sdc" ]'
oc-gluster-3.xxx glusterfs_ip=10.232.3.58 glusterfs_devices='[ "/dev/sdc" ]'
```

**Comment by xelfe** [comment]
@pgeorgiev333 Did you find something to make it work ?

**Comment by pgeorgiev333** [comment]
Yep. Here are the changes:

```
[OSEv3:children]
...
openshift_ca

[OSEv3:vars]
...
openshift_release=3.7
openshift_image_tag=v3.7.0
containerized=true
openshift_repos_enable_testing=true
...

[openshift_ca]
openshift.xxx
```

Then for every node i also added "containerized=true" , i.e.

```
[nodes]
openshift.xxx containerized=true openshift_schedulable=true openshift_node_labels="{'region': 'infra'}"
```

Honestly i'm not sure you need the "containerized=true" setting everywhere as i have not tested all other options without setting it but you can try and send feedback. I think the most important option is **"openshift_repos_enable_testing=true"** because i noticed that although i run the playbooks from the master (or release-3.7) branches i got 3.6.1 images pulled and 3.6.1 packages installed so that was the hint that helped me in the finding the right settings (mainly browsing google for the correct questions:) ).

After the install the openshift-ansible-service-broker deployment failed as it was looking for an image that does not exist. So I had to log on openshift, edit the deployment and update the image with the correct one: **registry.access.redhat.com/openshift3/ose-ansible-service-broker:latest** and deploy again.

Hope it helps.

**Comment by xelfe** [comment]
Well evrything was fixed with commit 2bf65cdab4aa88f160d005d3b7649b22a6dceba8  I think the issue can be close

**Comment by DanyC97** [comment]
@jarrpa can you close this issue?
